components/edit-collective/sections/Host.js

```diff
@@ -495,6 +496,16 @@ class Host extends React.Component {
             </Box>
           </Flex>
         </div>
+        {(selectedOption === 'selfHost' || selectedOption === 'ownHost') && (
+          <div>
+            <MessageBox type="info" fontSize="13px" withIcon mb={4}>
+              <FormattedMessage
+                id="collective.edit.host.legalName.info"
+                defaultMessage="Please set the legal name of the host in the Info section under Settings. This is required if your legal name is different than your display name for tax and accounting purposes."
```

Comment:
Would be great to have this as a link

![image](https://user-images.githubusercontent.com/1556356/132870365-bb1477b3-341a-40fc-8413-c6bf23aa7e80.png)

Reply:
@Betree : I've refactored based on your reviews. Now the info message appears only when the user has already chosen the self hosted options. Here's a screen-cast.

![legal-name-info-host](https://user-images.githubusercontent.com/12435965/132895323-90e9a8e3-e7d0-4f2b-9d48-feca6dbfc7c6.gif)

Reply:
Yes, I've done that. 👍🏽